### ISSUE TITLE
Changing log level

### ISSUE THREAD

**Issue by kaysond** [Jackett/Jackett#5650]
Is there a way to change the default log level? Right now the "info" level is obliterating my syslog

**Comment by garfield69** [comment]
are you a systemd user?  then #3723 may be relevant to you.

**Comment by kaysond** [comment]
Yes. I did see that while I was searching, but I was hoping there is an environment variable or something I can change to adjust it in the service file rather than configuring it in syslog?

**Comment by garfield69** [comment]
> is an environment variable or something I can change to adjust it in the service file rather than configuring it in syslog?

@flightlevel @kaso17  is there ?

**Comment by garfield69** [comment]
I'm going to guess not, since if there was the I would have expected #3723 to mention it.

**Comment by rrrevin** [comment]
Is Enhanced logging enabled? If so, disable it. I have non of the logging you have.
![image](https://user-images.githubusercontent.com/49560189/61174193-3f59cc00-a55a-11e9-99d1-34f667013f5c.png)

**Comment by kaysond** [comment]
its disabled
![image](https://user-images.githubusercontent.com/1147328/61175125-fc4e2780-a55e-11e9-8907-bd718693d86d.png)

**Comment by rrrevin** [comment]
Can you post your logs? I'm curious about the bt-scene Unknown indexer error

**Comment by garfield69** [comment]
that will most likely be from a call from Sonarr or similar app requesting from Jackett the use of an indexer which has since been deleted.
bt-scene was removed last month, bu the OP has not removed it from the app yet.

**Comment by kaysond** [comment]
^correct

**Comment by kaysond** [comment]
So just to be clear, log entries like this are not typical?
```
Jul 15 09:50:54 fserver jackett[23161]: 07-15 09:50:54 Info Found 25 (1 new) releases from KickAssTorrent (kat.li)
Jul 15 09:50:56 fserver jackett[23161]: 07-15 09:50:56 Info Found 20 (12 new) releases from ETTV
Jul 15 09:51:01 fserver jackett[23161]: 07-15 09:51:01 Info Found 4 (4 new) releases from The Pirate Bay
Jul 15 09:51:01 fserver jackett[23161]: 07-15 09:51:01 Info Found 50 (4 new) releases from 1337x
Jul 15 09:52:01 fserver jackett[23161]: 07-15 09:52:01 Info Found 50 (0 new) releases from 1337x
Jul 15 09:52:02 fserver jackett[23161]: 07-15 09:52:02 Info Found 20 (0 new) releases from ETTV
Jul 15 09:52:06 fserver jackett[23161]: 07-15 09:52:06 Info Found 19 (10 new) releases from KickAssTorrent (kat.li)
Jul 15 09:52:06 fserver jackett[23161]: 07-15 09:52:06 Info Found 3 (0 new) releases from The Pirate Bay
Jul 15 09:52:08 fserver jackett[23161]: 07-15 09:52:08 Info Found 50 (14 new) releases from EZTV
Jul 15 09:52:10 fserver jackett[23161]: 07-15 09:52:10 Info Found 67 (25 new) releases from Nyaa.si
Jul 15 09:52:12 fserver jackett[23161]: 07-15 09:52:12 Info Found 50 (11 new) releases from Anidex
Jul 15 09:52:21 fserver jackett[23161]: 07-15 09:52:21 Info Found 32 (2 new) releases from Horrible Subs
```

**Comment by garfield69** [comment]
perfectly normal entries seen in any Jackett log I've ever seen.

**Comment by kaysond** [comment]
Ok in that case I'd like to request an env var to change the log level to warning/error, etc. Or maybe even the option to log to a flat file rather than syslog?

**Comment by rrrevin** [comment]
ENV var is not the right way to do this , IMHO. As this is a service, I'd rather see a command line arg (or menu option in the GUI) for both syslog level and another for the syslog id/facility string used. You can then reconfigure syslog to redirect this to where you want.

**Comment by kaysond** [comment]
> ENV var is not the right way to do this , IMHO. 
I disagree. Plenty of services use environment variables to set log level. I just did this in Apache Guacamole, for example (https://guacamole.apache.org/doc/gug/guacamole-docker.html)

> As this is a service, I'd rather see a command line arg (or menu option in the GUI) for both syslog level and another for the syslog id/facility string used. You can then reconfigure syslog to redirect this to where you want.

Changing the syslog level or facility doesn't really address the problem, which is the sheer amount of unnecessary entries.

**Comment by rrrevin** [comment]
> Changing the syslog level or facility doesn't really address the problem, which is the sheer amount of unnecessary entries.

It solves it perfectly. The entries might be unnecessary perhaps for you, but necessary for others/debugging. By changing the syslog level (aka logging level), the code should automatically ONLY log events of that level and higher. This reduces entries by definition. If we need more details, we increase the logging level.

In your specific request, you'd set the syslog/logging level to "alert". You'll only see "alert/emerg" messages, and none of the "crit/err/warning/notice/info/debug"  messages etc. This solves both issues.

Here the key part: _Don't forget with syslog, You can redirect certain levels to /dev/null, where you'll **never** see them and it doesn't take up any disk._

**Comment by kaysond** [comment]
> By changing the syslog level (aka logging level), the code should automatically ONLY log events of that level and lower. 

Agreed. The way you'd written it made it sound like you want to globally change the level that's reported to syslog for all entries coming from jackett which didn't make any sense.

Changing jacketts log level is precisely what I'm looking for

**Comment by Nodens-** [comment]
Is there any particular reason the systemd unit does not come with LogLevelMax=3
as default?

**Comment by uheow847** [comment]
> > Changing the syslog level or facility doesn't really address the problem, which is the sheer amount of unnecessary entries.
> 
> It solves it perfectly. The entries might be unnecessary perhaps for you, but necessary for others/debugging. By changing the syslog level (aka logging level), the code should automatically ONLY log events of that level and lower. This reduces entries by definition. If we need more details, we increase the logging level.
> 
> In your specific request, you'd set the syslog/logging level to "alert". You'll only see "alert/emerg" messages, and none of the "crit/err/warning/notice/info/debug" messages etc. This solves both issues. https://en.wikipedia.org/wiki/Syslog#Severity_level
> 
> Here the key part: _Don't forget with syslog, you can redirect certain levels to /dev/null, where you'll **never** see them and it doesn't take up any disk._

I posted here but I see this is a related topic:
https://github.com/Jackett/Jackett/issues/3723

Essentially -- are there instructions to disable logging altogether?
